hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/TestAbsoluteResourceConfiguration.java

```diff
@@ -253,7 +253,7 @@ public void testSimpleMinMaxResourceConfigurartionPerQueue()
     rm.registerNode("127.0.0.1:1234", 250 * GB, 40);
 
     // Get queue object to verify min/max resource configuration.
-    CapacityScheduler cs = (CapacityScheduler) rm.getResourceScheduler();
+    CapacityScheduler cs = CapacitySchedulerTestUtilities.setupCapacityScheduler(rm, 250, 40);
```

Comment:
SetupCapacityScheduler is reused for getting a CS object and/or initing it. These could be separated to an init method and a factory method, so that there are no calls to the setup which doesn't use the return value. Or maybe the factory method could return a "MockCS" which inherits most of the production CS's methods, but calls the necessary init method after a `reinitialize` call. This way there is no need to call setupCapacityScheduler over and over.